postCheckReport.js

```diff
@@ -0,0 +1,252 @@
+    const { id, first_name, last_name, end_date } = employee;
+     if(end_date === null)
+    return { id, first_name, last_name };
+     else return null;
+  }).filter((element) => element !== null);
+
+  console.log('humi employees', employees.length);
+  ;
+  const timeOffUsers = timeOffUsersList.map((element) => {
+    const employee = element['attributes'];
+    const { employee_id, status, start_at, end_at } = employee;
+    return { employee_id, status, start_at, end_at };
+  });
+
+  // timeOffUsers don't have names associated with them, so we need to merge the two arrays on ID
+  // Pretty messy but works, maybe can refactor later
+
+  const timeOffUsersWithNames = employees.filter(({ id: id1 }) => timeOffUsers.some(({ employee_id: id2 }) => id2 === id1))
+  .map((element) => {
+    const { first_name, last_name } = element;
+    return {
+      realName: `${first_name} ${last_name}`,
+      ...element,
+    };
+  });
+
+
+  console.log('timeOffUsersWithNames',timeOffUsersWithNames)
+
+  // Filter the slack absent users
+
+  const humiNamesOnly = timeOffUsersWithNames.map(
+    (element) => element.realName
+  );
+  const absentUsersList = absentUsersWithDetail.filter((element) =>
+    !humiNamesOnly.includes(element.realName)
+  );
+  // Here are C-Level manager, they do not need to check-in everyday
+ const alwaysNotSendNotificationUsers = [
+      'Sam',
+      'Rohit Boolchandani',
+      'Shaun Jamieson',
+      'freshtimescounter',
+
+ ]
+ const finalAbsentUsersList = absentUsersList.filter((element) =>
+    !alwaysNotSendNotificationUsers.includes(element.realName )
+  );
+  // call helper func to eliminate all the user on vacation or on leave at here
+  console.log('test finalAbsentUsersList', finalAbsentUsersList);
+// in order to check if today is a holiday, we do a simple trick here,
+// if more than half employees does not show up, we suppose this is a holiday
+
+if(employees.length <=  finalAbsentUsersList.length * 2 ) {
+  console.log('Today is a holiday! no one is on');
+} else {
+
+    const text = `These user ${finalAbsentUsersList.map(
+        (coworker) => `<@${coworker.id}>`
+      )} still no sign up after send previous notification `;
+      await app.client.chat.postMessage({
+        token: process.env.SLACK_BOT_TOKEN,
+        channel: testChannelID,
+        text,
+      });
+    console.log('send post check notification');
+    }
+
+})();
+
+/** slack api */
+
+async function getAllUsers() {
+ const params = {
+    token: API_TOKEN,
+  };
+
+
+  const { data } = await axios.get(
+    'https://slack.com/api/users.list',
+    {
+      params,
+    },
+    config
+  );
+
+  return data.members;
+}
+
+async function getChannelUsers(channel) {
+  const params = {
+    token: API_TOKEN,
+    channel,
+  };
+
+
+
+  const { data } = await axios.get(
+    'https://slack.com/api/conversations.members',
+    {
+      params,
+    },
+    config
+  );
+  return data.members;
+}
+
+async function getMessages(channel) {
```

Comment:
Since _**latest**_ differs in both files by one hour, this function will have to take time params if we want to refactor it.